library/src/main/java/io/ona/kujaku/helpers/MapboxLocationComponentWrapper.java

```diff
@@ -0,0 +1,47 @@
+package io.ona.kujaku.helpers;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import com.mapbox.android.core.location.LocationEngine;
+import com.mapbox.mapboxsdk.location.LocationComponent;
+import com.mapbox.mapboxsdk.location.modes.CameraMode;
+import com.mapbox.mapboxsdk.location.modes.RenderMode;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+
+import io.ona.kujaku.exceptions.LocationComponentInitializationException;
+
+/**
+ * @author Vincent Karuri
+ */
+public class MapboxLocationComponentWrapper {
+
+    private static LocationComponent locationComponent;
```

Comment:
This will be prone to memory leaks where the map is closed. 
I think tracking if location component was initialized should be checked inside the KujakuMap view context via a flag and not as a static variable.

Reply:
I don't think I understand this concern about memory leakage. Could you rephrase?

Secondly, and as mentioned above, we should be trying to move functionality away from KujakuMapView?

Reply:
when mapview is closed it will not be reclaimed by gc because of this static variable which will still hold a reference to the mapview.

Reply:
The better design would be define `MapboxLocationComponentWrapper` as an a field in the KujakuMapView

Reply:
Fixed.